docs/core/compatibility/core-libraries/6.0/static-abstract-interface-methods.md

```diff
@@ -0,0 +1,44 @@
+---
+title: "Breaking change: Static abstract members in interfaces"
+description: Learn about the .NET 6.0 breaking change where `static` interface members can now be marked `abstract`.
+ms.date: 09/08/2021
+---
+# Static abstract members in interfaces
+
+.NET 6 is previewing a new feature where `static` interface members can be marked as `abstract`. This feature involves several changes to the ECMA 335 spec to allow intermediate language (IL) metadata patterns that were previously considered illegal. For more information, see [dotnet/runtime#49558](https://github.com/dotnet/runtime/issues/49558).
+
+> [!NOTE]
+> The feature is in preview, and it's possible that additional changes or support may be introduced between .NET 6 and a future version of .NET when the feature stabilizes.
+
+## Old behavior
+
+Static interface members could not be marked as `abstract`. Marking them `abstract` resulted in [Compiler error CS0112](../../../../csharp/misc/cs0112.md) and would generate illegal IL.
```

Comment:
In my opinion, the statement is confusing. An error reported by C# compiler prevents any IL generation. what would be generated is undefined. Behavior of the compiler and validity of metadata are two different things and probably should not be conflated.

Reply:
This isn't just about C#'s view. This is about downstream tooling interpreting metadata running into issues.

Prior to this runtime feature, such IL was considered invalid and so tools and other compilers would fail when encountering it. Now it is valid and tools/compilers need to be updated to account for it.

C++/CLI and Powershell were both broken because of this and requested the breaking change be documented.

Reply:
> This isn't just about C#'s view. This is about downstream tooling interpreting metadata running into issues.

I understand. I am saying that these are different things. and should be described separately. What C# behavior used to be, what IL was considered illegal, etc. C# simply wouldn't emit anything in case of the error. How one is supposed to guess what IL would be illegal based on that information.

Reply:
That makes sense. I didn't fully process the initial comment.